xbmc/windowing/gbm/drm/DRMAtomic.cpp

```diff
@@ -60,22 +46,18 @@ uint32_t GetScalingFactor(uint32_t srcWidth,
 
 bool CDRMAtomic::SetScalingFilter(struct CDRMObject* object, const char* name, const char* type)
 {
-  uint32_t filter_type = GetScalingFilterType(type);
-  if (object->SupportsPropertyAndValue(name, filter_type))
+  if (AddPropertyValueByName(object, name, type))
   {
-    if (AddProperty(object, name, filter_type))
-    {
-      uint32_t mar_scale_factor =
-          GetScalingFactor(m_width, m_height, m_mode->hdisplay, m_mode->vdisplay);
-      uint32_t diff_w = m_mode->hdisplay - (mar_scale_factor * m_width);
-      uint32_t diff_h = m_mode->vdisplay - (mar_scale_factor * m_height);
-      AddProperty(object, "CRTC_X", (diff_w / 2));
-      AddProperty(object, "CRTC_Y", (diff_h / 2));
-      AddProperty(object, "CRTC_W", (mar_scale_factor * m_width));
-      AddProperty(object, "CRTC_H", (mar_scale_factor * m_height));
-      return true;
+    uint32_t mar_scale_factor =
+        GetScalingFactor(m_width, m_height, m_mode->hdisplay, m_mode->vdisplay);
+    uint32_t diff_w = m_mode->hdisplay - (mar_scale_factor * m_width);
+    uint32_t diff_h = m_mode->vdisplay - (mar_scale_factor * m_height);
+    AddProperty(object, "CRTC_X", (diff_w / 2));
+    AddProperty(object, "CRTC_Y", (diff_h / 2));
+    AddProperty(object, "CRTC_W", (mar_scale_factor * m_width));
+    AddProperty(object, "CRTC_H", (mar_scale_factor * m_height));
```

Comment:
Btw. what will happen if DRM rejects these CRTC coordinates? That is, KMS atomic commit fails.

Not all planes might support values different from (0, 0, hdisplay, vdisplay).

Reply:
The idea here is to match the way GL is doing to make frame centralize after upscaling. I am not sure if there are some planes do not allow to request drawing from position other than (0, 0).

Reply:
Traditionally all primary planes refuse if you don't fill the whole plane exactly with the FB.

Any driver could refuse it for any reason, so there needs to be a fallback, but that applies to atomic KMS in general.

Reply:
I got it. With that change user experience may not be good I thought. Upscaled frame will be drawn on left top corner and right bottom corner will be painted black.
Attached the difference in the user experience for your reference. 
![image](https://user-images.githubusercontent.com/25810508/96134674-f50f8980-0f18-11eb-9a76-34a8c192995a.png)
I felt right one is better in viewing the game content so I have added that code. 
We can do one check if 
` AddProperty(object, "CRTC_X", (diff_w / 2));
  AddProperty(object, "CRTC_Y", (diff_h / 2));` 
fail then we can set to (0, 0).  What do you suggest ?

Reply:
ok. I will keep  CRTC (X, Y) to (0, 0) as it was before.

Reply:
No, that is not what I meant.

Setting the CRTC X, Y, W, H to center the integer-scaled image is fine!

However, you **must** prepare for the fact that the KMS driver may not be able to do that, so you need a *fallback* path.